src/caip-chaid-id.test-d.ts

```diff
@@ -0,0 +1,23 @@
+import { expectAssignable, expectNotAssignable } from 'tsd';
+
+import { CaipChainId } from '.';
+
+// Valid caip chain id strings:
+
+expectAssignable<CaipChainId>('namespace:reference');
+
+expectAssignable<CaipChainId>('namespace:');
+
+expectAssignable<CaipChainId>(':reference');
+
+const embeddedString = 'test';
+expectAssignable<CaipChainId>(`${embeddedString}:${embeddedString}`);
+
+// Not valid caip chain id strings:
+
+expectAssignable<CaipChainId>('namespace:😀');
+expectAssignable<CaipChainId>('😀:reference');
```

Comment:
should these be above the 'Not valid caip chain id strings:' comment?

Reply:
technically the emojis are strings, so these are valid